packages/manager/modules/vps/src/dashboard/cancel-resiliation/vps-cancel-resiliation.routing.js

```diff
@@ -0,0 +1,88 @@
+import find from 'lodash/find';
+import { BillingService } from '@ovh-ux/manager-models';
+
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('vps.detail.dashboard.cancel-resiliation', {
+    url: '/cancel-resiliation',
+    views: {
+      modal: {
+        component: 'billingCancelResiliation',
+      },
+    },
+    layout: 'modal',
+    translations: { value: ['.'], format: 'json' },
+    resolve: {
+      cancelResiliation: /* @ngInject */ (
+        $q,
+        engagement,
+        OvhApiBillingAutorenewServices,
+        setReactivateEngagementStrategy,
+      ) => (service) => {
+        if (engagement) {
+          return setReactivateEngagementStrategy();
+        }
+        service.cancelResiliation();
+        return OvhApiBillingAutorenewServices.Aapi()
+          .put(
+            {},
+            {
+              updateList: [service],
+            },
+          )
+          .$promise.then((result) => {
+            if (result.state === 'OK') {
+              return result;
+            }
+            return $q.reject(result);
+          });
+      },
+      endStrategies: /* @ngInject */ (endStrategyEnum) =>
+        endStrategyEnum.reduce(
+          (object, strategy) => ({
+            ...object,
+            [strategy]: strategy,
+          }),
+          {},
+        ),
+      endStrategyEnum: /* @ngInject */ ($http) =>
+        $http
+          .get('/services.json')
+          .then(
+            ({ data }) =>
+              data.models['services.billing.engagement.EndStrategyEnum']?.enum,
+          ),
```

Comment:
This looks like it could have been handled by the component as call looks generic to all products (`/services` API) but this is due to the component design

Reply:
Yeah... Agreed